### ISSUE TITLE
Allow filtering by multiple repositories with search prefix

### ISSUE THREAD

**Issue by andrew** [octobox/octobox#919]
Feature requested by @gbp.

For a search of `repo:24pullrequests/24pullrequests,octobox/octobox`, should return results for both repositories. Currently ignores any more after the first repository.

Two bits of code need changing for this, the easy one, remove `.first` from https://github.com/octobox/octobox/blob/master/app/models/search.rb#L72

and the second, harder one, on https://github.com/octobox/octobox/blob/master/app/models/notification.rb#L33 we need to convert an array of strings (of any length) like `["24pullrequests/24pullrequests", "octobox/octobox"]` into an arel query like:

```ruby
Notification.arel_table[:repository_full_name].matches("24pullrequests/24pullrequests"').or(Notification.arel_table[:repository_full_name].matches("octobox/octobox"))
```

Chaining extra `.or(Notification.arel_table[:repository_full_name].matches(repo_full_name)` on the end for every extra repo name in the argument.

n.b. we're using `arel_table[:field].matches` because it switches between LIKE or ILIKE depending on if postgres or mysql is being used.

**Comment by prashuchaudhary** [comment]
@andrew I have made the recommended changes. Please have a look

**Comment by andrew** [comment]
Fixed by #921, thanks @prashuchaudhary!
